dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageErrorDiagnosis.java

```diff
@@ -91,4 +106,76 @@ public final String toString() {
 
     return builder.toString();
   }
+
+  /** Returns a map from the cause of linkage errors to the diagnosis. */
+  public static ImmutableMap<LinkageErrorCause, LinkageErrorDiagnosis> diagnoseJarLinkageReport(
+      JarLinkageReport jarLinkageReport, Map<Path, Artifact> pathToArtifact)
+      throws RepositoryException, IOException {
+    ImmutableMap.Builder<LinkageErrorCause, LinkageErrorDiagnosis> causeToDiagnosis =
+        ImmutableMap.builder();
+    Artifact sourceArtifact = pathToArtifact.get(jarLinkageReport.getJarPath());
+
+    ImmutableMultimap<LinkageErrorCause, String> causeToSourceClasses =
+        jarLinkageReport.getCauseToSourceClasses();
+    if (causeToSourceClasses.isEmpty()) {
+      return causeToDiagnosis.build();
+    }
+
+    ImmutableMap<String, Artifact> selectedArtifacts =
+        Maps.uniqueIndex(pathToArtifact.values(), Artifacts::makeKey);
+
+    ImmutableMap<Path, Artifact> pathToArtifactForJar =
+        ClassPathBuilder.getPathToArtifact(ImmutableList.of(sourceArtifact));
+
+    for (LinkageErrorCause cause : causeToSourceClasses.keySet()) {
+      switch (cause.getReason()) {
+        case CLASS_NOT_FOUND:
+          LinkageErrorDiagnosis diagnosis =
+              diagnoseMissingClass(cause, sourceArtifact, selectedArtifacts, pathToArtifactForJar);
+          causeToDiagnosis.put(cause, diagnosis);
+          break;
+        default:
+          // TODO: Implement other cases (SYMBOL_NOT_FOUND, INACCESSIBLE_CLASS, etc.)
```

Comment:
probably worth filling in the switch statement now even if the methods are no-ops for the time being.